xarray/backends/rasterio_.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+
+try:
+    import rasterio
+except ImportError:
+    rasterio = False
+
+from .. import Variable
+from ..core.utils import FrozenOrderedDict, Frozen, NDArrayMixin, is_scalar
+from ..core import indexing
+from ..core.pycompat import OrderedDict, suppress
+
+from .common import AbstractDataStore
+
+_rio_varname = 'raster'
+
+_error_mess = ('The kind of indexing operation you are trying to do is not '
+               'valid on RasterIO files. Try to load your data with ds.load()'
+               'first.')
+
+
+class RasterioArrayWrapper(NDArrayMixin):
+    def __init__(self, ds):
+        self.ds = ds
+        self._shape = self.ds.count, self.ds.height, self.ds.width
+        self._ndims = len(self.shape)
+
+    @property
+    def dtype(self):
+        return np.dtype(self.ds.dtypes[0])
+
+    @property
+    def shape(self):
+        return self._shape
+
+    def __getitem__(self, key):
+
+        # make our job a bit easier
+        key = indexing.canonicalize_indexer(key, self._ndims)
+
+        # bands cannot be windowed but they can be listed
+        bands, n = key[0], self.shape[0]
```

Comment:
Put this on two lines.

Use more descriptive variable names, e.g., `bands_key` and `num_bands`